src/components/ui-shell/_side-nav.scss

```diff
@@ -390,6 +390,18 @@
     fill: $shell-side-nav-icon-01;
   }
 
+  .#{$prefix}--side-nav__icon > svg[data-side-nav-collapse-icon] {
```

Comment:
Do we need to add these `data-*` attributes to the selector here? If so, each implementation would also have to add these in, right?

Reply:
we want to keep the data attributes here in this case so that we do not render both icons at the same time. Yes, they will be required in any implementation

Reply:
Wait, I don't understand why would these data attributes be needed for any implementation? Seems like we shouldn't force data attributes on other frameworks for this unless we absolutely have to. Could we add a class name to help out here, instead?

Reply:
You're right they don't necessarily need to be data attributes. As you suggested, using classes would accomplish the same thing. The only reason why I stuck with data attributes was for consistency, as it seems we use this approach in some of our other components like text-input and carousel. Aside from remaining consistent with our implementation throughout the library, I have no preference towards either approach